src/types/mod.rs

```diff
@@ -204,6 +204,60 @@ pub use self::capabilities::Capabilities;
 /// re-exported from imap_proto;
 pub use imap_proto::StatusAttribute;
 
+
+// We need a ResponseCode that is not tied to a lifetime, to be used in UnsolicitedResponse.
+/// Response code that may be sent with OK/NO/BAD/BYE responses.
+/// See [RFC 3501](https://tools.ietf.org/html/rfc3501#section-3.1).
+#[derive(Debug, Eq, PartialEq)]
+pub enum ResponseCode {
+    //Alert: not parsed by imap-proto yet.
+    //BadCharset: not parsed by imap-proto yet.
+    //Capability: not parsed by imap-proto yet.
+    //Parse: not parsed by imap-proto yet.
```

Comment:
Would be good to link to an imap-proto issue for this.

Reply:
Yes, I'm working on a pull request for that.